### ISSUE TITLE
Feature Request: Exclude hidden entities

### ISSUE THREAD

**Issue by klatka** [Hypfer/hass-scene_presets#28]
### What is happening?

When I select an area and apply a scene, all the lights assigned to the area undergo a color change. However, there are also lights that are hidden and should not change their color (This is the normal behavior of Home Assistant. If an area is selected in a service call, hidden objects are also included).

### What should be happening?

A checkbox to exclude hidden entities if you select a device or an area would be a great enhancement.

### Reflection 1

Objective Input

### Is this issue tracker the right place for this topic?

Yes

### Please enter the scene_preset version you're using

2.2.1

### Please enter the Home Assistant version you're using

2024.8.2

### Details about your setup

Various integrations

### Reflection 2

-

### Time and Effort

- [X] Yes, I did my homework.
- [X] I understand that my issue will be deleted with no further comment if I did not do my homework.

**Comment by Hypfer** [comment]
> A checkbox to exclude hidden entities if you select a device or an area would be a great enhancement.

Nope, it would not, because that's how you get incomprehensible and convoluted UIs that are no fun to use.
"Just add a checkbox/a toggle/a switch for this behavioral change" is pretty much the default when people just did not think about the bigger picture.

Problem is that I can't think about the bigger picture either, because I have no clue what you want to achieve.
I just know that you want a checkbox for a behavioral change, but that is already a solution to a problem that isn't mentioned. With that, it's impossible to find a better solution that isn't just a new checkbox, because for that, the problem needs to be known.

So. What is the problem? What is your setup? Why are there areas with hidden entities?
Why should they be ignored? But why only sometimes?

**Comment by klatka** [comment]
It's not always necessary to dig into the bigger picture if the goal is to align with the functionality the Home Assistant platform already supports. The question of why there should be hidden entities has already been addressed by Home Assistant's decision to include this feature.

Let me provide more information about my problem giving one example: I have an Ulanzi Pixel Clock that is connected via MQTT and this device is assigned to an area in Home Assistant. The issue I am facing is that the Pixel Clock exposes a light entity to adjust the color of the content the Pixel Clock is displaying. I don't want the applied preset to change the content's color. I also don't want to show this light entity in my Lighting UI. To accomplish that I simply hid this light entity and I was expecting that the Scene Preset also ignore this light entity when selecting an area for the preset.
